apps/client/src/auth/auth-service.ts

```diff
@@ -14,6 +14,31 @@ class CognitoAuthService implements AuthService {
     const session = await Auth.currentSession();
     return session.getAccessToken().getJwtToken();
   }
+
+  /**
+   * This function accepts a callback function to call when application is signed-in or already signed-in
+   * @param callback the callback function to call when application is signed-in
+   */
+  async onSignedIn(callback: () => unknown) {
+    /**
+     * Either Auth.currentAuthenticatedUser() or callback of Hub.listen('auth', xxx) is executed initially;
+     * Callback of Hub.listen('auth', xxx) is executed for every sign-in;
+     */
+    try {
+      // Check for initial authentication state
+      await Auth.currentAuthenticatedUser();
+      callback();
+    } catch (e) {
+      // NOOP; not yet authenticated;
```

Comment:
worth adding an error log ?

Reply:
I'm hesitant to add an error log here since it is the expected behavior before authentication;